examples/common/python/utility/signature.py

```diff
@@ -230,9 +233,13 @@ def generate_client_signature(self, input_json_str,
         encrypted_session_key_str = ''.join(format(i, '02x') for i in encrypted_session_key)
         self.__encrypt_workorder_indata(input_json_params, session_key,
                 session_iv, worker.encryption_key, data_key, data_iv)
-        # [NO_OF_BYTES] 16 BYTES for nonce, is the recommendation by NIST to
-        # avoid collisions by the "Birthday Paradox".
-        nonce =  crypto.random_bit_string(NO_OF_BYTES)
+
+        if input_json_params["requesterNonce"]:
+            nonce = crypto.string_to_byte_array(input_json_params["requesterNonce"])
```

Comment:
Are we checking elsewhere if input_json_params["requesterNonce"] is of type hex string. will crypto.string_to_byte_array() throw exception if input is not string. Then its good to catch the exception

Reply:
Ok, i added a check to validate hex string